<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8">
  <meta name="generator" content="pandoc">
  <meta name="author" content="Shaun Jackman" />
  <meta name="dcterms.date" content="2015-07-27" />
  <title>PhD thesis committee meeting</title>
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black-translucent" />
  <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no">
  <link rel="stylesheet" href="reveal.js/css/reveal.css"/>
  <style type="text/css">code{white-space: pre;}</style>
  <link rel="stylesheet" href="reveal.js/css/theme/sky.css" id="theme">
  <!-- If the query includes 'print-pdf', include the PDF print sheet -->
  <script>
    if( window.location.search.match( /print-pdf/gi ) ) {
      var link = document.createElement( 'link' );
      link.rel = 'stylesheet';
      link.type = 'text/css';
      link.href = 'reveal.js/css/print/pdf.css';
      document.getElementsByTagName( 'head' )[0].appendChild( link );
    }
  </script>
  <!--[if lt IE 9]>
  <script src="reveal.js/lib/js/html5shiv.js"></script>
  <![endif]-->
</head>
<body>
  <div class="reveal">
    <div class="slides">

<section>
    <h1 class="title">PhD thesis committee meeting</h1>
    <h2 class="author">Shaun Jackman</h2>
    <h3 class="date">2015-07-27</h3>
</section>

<section><section id="people" class="titleslide slide level1"><h1>People</h1></section><section id="shaun-jackman" class="slide level2">
<h1>Shaun Jackman</h1>
<p><a href="http://bcgsc.ca">Genome Sciences Centre</a>, BC Cancer Agency<br />
Vancouver, Canada<br />
<a href="https://twitter.com/sjackman">@sjackman</a><br />
<a href="https://github.com/sjackman">github.com/sjackman</a><br />
<a href="http://sjackman.ca">sjackman.ca</a></p>
<p><a href="http://sjackman.ca"><img src="images/sjackman.jpg" alt="Photo" /></a></p>
</section><section id="thesis-committee" class="slide level2">
<h1>Thesis committee</h1>
<p><a href="http://www.bcgsc.ca/faculty/inanc-birol">Inanc Birol</a><br />
<a href="http://bohlmannlab.msl.ubc.ca/">Joerg Bohlmann</a><br />
<a href="http://www.stat.ubc.ca/~jenny/">Jenny Bryan</a><br />
<a href="http://hallam.microbiology.ubc.ca/">Steven Hallam</a></p>
<p> </p>
<h3 id="chair">Chair</h3>
<p><a href="http://www.bcgsc.ca/faculty/sjones">Steven Jones</a></p>
</section></section>
<section><section id="activities-in-2015" class="titleslide slide level1"><h1>Activities in 2015</h1></section><section class="slide level2">

<h3 id="conferences">Conferences</h3>
<p><a href="http://unconf.ropensci.org/">rOpenSci Unconference</a><br />
<a href="http://f1000.com/posters/browse/summary/1097878">RECOMB 2015</a></p>
<h3 id="teaching-assistant">Teaching assistant</h3>
<p><a href="http://stat545-ubc.github.io/">STAT 545</a> Data wrangling, exploration, and analysis with R<br />
Automating data analysis pipelines</p>
<p><a href="https://courses.students.ubc.ca/cs/main?dept=BIOF&amp;course=520">BIOF 520</a> Problem-Based Learning In Bioinformatics<br />
Genomic epidemiology</p>
</section><section class="slide level2">

<ul>
<li><a href="http://journals.plos.org/plosone/article?id=10.1371/journal.pone.0128026">UniqTag</a>: Content-derived unique and stable identifiers for gene annotation<br />
<small> <strong>SD Jackman</strong>, J Bohlmann, I Birol<br />
PLOS ONE 2015 </small></li>
<li>Sealer: a scalable gap-closing application for finishing draft genomes<br />
<small> D Paulino, RL Warren, BP Vandervalk, A Raymond, <strong>SD Jackman</strong>, I Birol<br />
BMC Bioinformatics 2015 </small></li>
<li>Improved white spruce (<em>Picea glauca</em>) genome assemblies and annotation of large gene families of conifer terpenoid and phenolic defense metabolism<br />
<small> RL Warren, CI Keeling, MMS Yuen, A Raymond, GA Taylor, …<br />
The Plant Journal 2015 </small></li>
<li>Spaced Seed Data Structures for De Novo Assembly<br />
<small> I Birol, J Chu, H Mohamadi, <strong>SD Jackman</strong>, K Raghavan, BP Vandervalk, …<br />
International Journal of Genomics 2015 </small></li>
<li>DIDA: Distributed Indexing Dispatched Alignment<br />
<small> H Mohamadi, BP Vandervalk, A Raymond, <strong>SD Jackman</strong>, J Chu, …<br />
PLOS ONE 2015 </small></li>
<li>On the Representation of De Bruijn Graphs<br />
<small> R Chikhi, A Limasset, <strong>SD Jackman</strong>, JT Simpson, P Medvedev<br />
Journal of Computational Biology 2015 </small></li>
<li>BioBloom tools: fast, accurate and memory-efficient host species sequence screening using bloom filters<br />
<small> J Chu, S Sadeghi, A Raymond, <strong>SD Jackman</strong>, KM Nip, R Mar, …<br />
Bioinformatics 2014 </small></li>
<li>Konnector: Connecting paired-end reads using a bloom filter de Bruijn graph<br />
<small> BP Vandervalk, <strong>SD Jackman</strong>, A Raymond, H Mohamadi, C Yang, D Attali, …<br />
Bioinformatics and Biomedicine (BIBM) 2014 </small></li>
</ul>
</section></section>
<section><section id="manuscripts" class="titleslide slide level1"><h1>Manuscripts</h1></section><section id="white-spruce-organelles" class="slide level2">
<h1>White Spruce Organelles</h1>
<p>Organellar Genomes of White Spruce (<em>Picea glauca</em>): Assembly and Annotation</p>
<p><a href="https://github.com/sjackman/white-spruce-organelle-paper"><img src="images/plastid-annotation.png" alt="Plastid" /> <img src="images/mt-annotation.png" alt="Mitochondrion" /></a></p>
</section><section id="abyss" class="slide level2">
<h1>ABySS</h1>
<h3 id="assembly-by-spaced-seeds">Assembly by Spaced Seeds</h3>
<p>for the assembly of long reads</p>
<p><a href="https://github.com/bcgsc/abyss-paired-dbg-paper"><img src="images/ABySS-2.0.png" alt="ABySS 2.0" /></a></p>
</section><section id="distanceest" class="slide level2">
<h1>DistanceEst</h1>
<p>Estimating the distance between two contigs</p>
<p><a href="https://github.com/sjackman/distance-estimate-paper"><img src="images/DistanceEst.png" alt="DistanceEst" /></a></p>
</section><section id="homebrew-science" class="slide level2">
<h1>Homebrew Science</h1>
<p><a href="http://brew.sh">Homebrew</a> | <a href="http://brew.sh/linuxbrew/">Linuxbrew</a> | <a href="http://brew.sh/homebrew-science/">Homebrew-science</a></p>
<p><a href="http://brew.sh/homebrew-science/"><img src="images/homebrew-science.png" alt="Homebrew-science" /></a></p>
<p><a href="http://sjackman.github.io/open-science/#/homebrew-navigates-dependency-hell"><img src="images/homebrew-bioinformatics.png" alt="Dependencies of bioinformatics tools in Homebrew" /></a></p>
</section></section>
<section><section id="thesis-proposal" class="titleslide slide level1"><h1>Thesis proposal</h1></section><section id="thesis-proposal-1" class="slide level2">
<h1>Thesis proposal</h1>
<ol type="1">
<li>Organellar Genomes of White Spruce (<em>Picea glauca</em>): Assembly and Annotation</li>
<li>Scale ABySS to assemble long, accurate reads</li>
<li>Scaffold using SMS and molecular barcoding</li>
<li>Assemble single-molecule sequencing data</li>
</ol>
</section></section>
<section><section id="organellar-genomes-of-white-spruce-assembly-and-annotation" class="titleslide slide level1"><h1>Organellar Genomes of White Spruce: Assembly and Annotation</h1></section><section id="organellar-genomes" class="slide level2">
<h1>Organellar genomes</h1>
<h3 id="completed">Completed</h3>
<ul>
<li>Assembled cpDNA and mtDNA genomes</li>
<li>Annotated genes (mRNA, rRNA, tRNA) and repeats</li>
<li>Analysed RNA-seq data to quantify
<ul>
<li>transcript abundance in eight tissues</li>
<li>expressed ORFs</li>
<li>C-to-U RNA editing</li>
</ul></li>
</ul>
</section><section id="organellar-genomes-1" class="slide level2">
<h1>Organellar genomes</h1>
<h3 id="plan">Plan</h3>
<ul>
<li>Analyse RNA-seq data to annotate
<ul>
<li>cryptic ACG start codons<br />
 due to C-to-U RNA editing</li>
<li>spliced ORFs</li>
</ul></li>
<li>Submit annotated genomes to GenBank</li>
<li>Complete manuscript</li>
</ul>
</section></section>
<section><section id="scale-abyss-to-assemble-long-accurate-reads" class="titleslide slide level1"><h1>Scale ABySS to assemble long, accurate reads</h1></section><section id="scaling-abyss" class="slide level2">
<h1>Scaling ABySS</h1>
<h3 id="for-the-assembly-of-long-reads">For the assembly of long reads</h3>
<ul>
<li>500 bp overlapping paired-end reads (MiSeq)</li>
<li>10 kbp synthetic long reads (Moleculo)</li>
<li>8–200 kbp corrected single-molecule sequencing</li>
</ul>
<p> </p>
<h3 id="using-memory-efficient-data-structures">Using memory efficient data structures</h3>
<ul>
<li>spaced seed de Bruijn Graph</li>
<li>Bloom filter de Bruijn Graph</li>
</ul>
</section><section id="scaling-abyss-1" class="slide level2">
<h1>Scaling ABySS</h1>
<h3 id="completed-1">Completed</h3>
<ul>
<li>Implemented Bloom filter de Bruijn Graph
<ul>
<li><a href="http://dx.doi.org/10.1109/BIBM.2014.6999126">Konnector</a> with Ben</li>
<li><a href="http://www.biomedcentral.com/1471-2105/16/230">Sealer</a> with Daniel</li>
</ul></li>
<li>Implemented spaced seed de Bruijn Graph with Karthika</li>
<li>Reduced memory and time requirements for long reads</li>
<li>Memory usage is independent of parameter <em>k</em></li>
</ul>
<h3 id="plan-1">Plan</h3>
<ul>
<li>Assemble model organisms, human in particular, with ABySS 2.0</li>
<li>Compare to ABySS 1.5 and other dBG assemblers</li>
</ul>
</section><section id="recomb-2015" class="slide level2">
<h1>RECOMB 2015</h1>
<p>Research in Computational Molecular Biology<br />
Warsaw, Poland · 2015 April 10–15</p>
<p><a href="http://f1000.com/posters/browse/summary/1097878"><img src="images/recomb-2015-poster.png" alt="RECOMB 2015 poster" /></a></p>
</section></section>
<section><section id="scaffold-using-sms-and-molecular-barcoding" class="titleslide slide level1"><h1>Scaffold using SMS and molecular barcoding</h1></section><section id="scaffolding" class="slide level2">
<h1>Scaffolding</h1>
<p>Order and orient contigs to build scaffolds using…</p>
<p> </p>
<h3 id="completed-2">Completed</h3>
<ul>
<li>Illumina mate-pair libraries</li>
<li>Illumina Moleculo with Tony</li>
</ul>
<p> </p>
<h3 id="plan-2">Plan</h3>
<ul>
<li>Single-molecule sequencing</li>
<li>Molecular barcoding<br />
 10X Genomics GemCode<br />
 for white spruce (<em>Picea glauca</em>)</li>
</ul>
</section></section>
<section><section id="assemble-single-molecule-sequencing-data" class="titleslide slide level1"><h1>Assemble single-molecule sequencing data</h1></section><section id="assemble-sms" class="slide level2">
<h1>Assemble SMS</h1>
<h3 id="drosophila-melanogaster"><em>Drosophila melanogaster</em></h3>
<ul>
<li>Assembly with <a href="http://www.cbcb.umd.edu/software/PBcR/">PBcR</a> used 621,000 CPU-hours,<br />
 or 26 days with 1000 cores</li>
<li><a href="https://github.com/marbl/MHAP">MHAP</a> reduced assembly time to 700 CPU-hours</li>
<li>Efficient algorithms make a big difference!</li>
</ul>
<p> </p>
<h3 id="homo-sapiens"><em>Homo sapiens</em></h3>
<ul>
<li>Assembly of <a href="http://blog.pacificbiosciences.com/2014/02/data-release-54x-long-read-coverage-for.html">CHM1</a> with <a href="https://github.com/PacificBiosciences/Bioinformatics-Training/wiki/HGAP-in-SMRT-Analysis">HGAP</a> used 405,000 CPU-hours<br />
 in one day on Google Cloud!</li>
<li>More work to be done for PacBio</li>
<li>And lots more work to be done for Nanopore</li>
</ul>
</section><section id="assemble-sms-1" class="slide level2">
<h1>Assemble SMS</h1>
<h3 id="overlap">Overlap</h3>
<ul>
<li>Spaced seed clustering with Hamid and Justin</li>
</ul>
<h3 id="layout">Layout</h3>
<ul>
<li>Numerical solution with ABySS-DrawGraph</li>
</ul>
<h3 id="consensus">Consensus</h3>
<ul>
<li>BWA-MEM and samtools</li>
<li>Assembled a small 100 kbp synthetic genome</li>
</ul>
<h3 id="plan-3">Plan</h3>
<ul>
<li>Continue development with Hamid, Justin and Ben</li>
<li>Assemble Nanopore <em>Escherichia coli</em> (5 Mbp)
<ul>
<li>Nanopore <em>Saccharomyces cerevisiae</em> (12 Mbp)</li>
<li>PacBio <em>Homo sapiens</em> <a href="http://blog.pacificbiosciences.com/2014/02/data-release-54x-long-read-coverage-for.html">CHM1</a> (3 Gbp)</li>
</ul></li>
</ul>
</section></section>
<section><section id="thesis-proposal-2" class="titleslide slide level1"><h1>Thesis proposal</h1></section><section id="thesis-proposal-3" class="slide level2">
<h1>Thesis proposal</h1>
<ol type="1">
<li>Organellar Genomes of White Spruce (<em>Picea glauca</em>): Assembly and Annotation</li>
<li>Scale ABySS to assemble long, accurate reads</li>
<li>Scaffold using SMS and molecular barcoding</li>
<li>Assemble single-molecule sequencing data</li>
</ol>
</section></section>
<section><section id="fin" class="titleslide slide level1"><h1>fin</h1></section><section id="shaun-jackman-1" class="slide level2">
<h1>Shaun Jackman</h1>
<p><a href="http://bcgsc.ca">Genome Sciences Centre</a>, BC Cancer Agency<br />
Vancouver, Canada<br />
<a href="https://twitter.com/sjackman">@sjackman</a><br />
<a href="https://github.com/sjackman">github.com/sjackman</a><br />
<a href="http://sjackman.ca">sjackman.ca</a></p>
<p><a href="http://sjackman.ca"><img src="images/sjackman.jpg" alt="Photo" /></a></p>
</section><section id="links" class="slide level2">
<h1>Links</h1>
<p><a href="http://www.bcgsc.ca/faculty/inanc-birol">Inanc Birol</a> | <a href="http://bohlmannlab.msl.ubc.ca/">Joerg Bohlmann</a><br />
<a href="http://hallam.microbiology.ubc.ca/">Steven Hallam</a> | <a href="http://www.stat.ubc.ca/~jenny/">Jenny Bryan</a><br />
<a href="http://f1000.com/posters/browse/summary/1097878">RECOMB 2015</a> | <a href="http://unconf.ropensci.org/">rOpenSci Unconference</a><br />
<a href="http://journals.plos.org/plosone/article?id=10.1371/journal.pone.0128026">UniqTag</a><br />
<a href="https://github.com/sjackman/white-spruce-organelle-paper">White spruce organelles</a> | <a href="https://github.com/bcgsc/abyss-paired-dbg-paper">ABySS 2.0</a> | <a href="https://github.com/sjackman/distance-estimate-paper">DistanceEst</a><br />
<a href="https://courses.students.ubc.ca/cs/main?dept=BIOF&amp;course=520">BIOF 520</a> | <a href="http://stat545-ubc.github.io/">STAT 545</a><br />
<a href="http://sjackman.github.io/open-science/#/homebrew-navigates-dependency-hell">Open, reproducible science</a><br />
<a href="http://brew.sh">Homebrew</a> | <a href="http://brew.sh/linuxbrew/">Linuxbrew</a> | <a href="http://brew.sh/homebrew-science/">Homebrew-science</a></p>
</section><section id="software" class="slide level2">
<h1>Software</h1>
<p><a href="https://github.com/PacificBiosciences/blasr">BLASR</a> | <a href="https://github.com/lh3/bwa">BWA-MEM</a> | <a href="http://wgs-assembler.sourceforge.net/">Celera Assembler</a><br />
<a href="https://github.com/thegenemyers/DALIGNER">DALIGNER</a> | <a href="https://dazzlerblog.wordpress.com/">Dazzler</a> | <a href="https://github.com/PacificBiosciences/falcon">Falcon</a> | <a href="https://github.com/PacificBiosciences/Bioinformatics-Training/wiki/HGAP-in-SMRT-Analysis">HGAP</a><br />
<a href="http://last.cbrc.jp/">LAST</a> | <a href="https://github.com/marbl/MHAP">MHAP</a> | <a href="http://schatzlab.cshl.edu/data/nanocorr/">Nanocorr</a> | <a href="https://github.com/jts/nanocorrect">Nanocorrect</a><br />
<a href="https://github.com/jts/nanopolish">Nanopolish</a> | <a href="http://www.cbcb.umd.edu/software/PBcR/">PBcR</a> | <a href="https://github.com/PacificBiosciences/pbdagcon">PBDAGCon</a> | <a href="https://sourceforge.net/projects/poamsa/">POA</a><br />
<a href="https://github.com/PacificBiosciences/GenomicConsensus">Quiver</a></p>
</section></section>
<section><section id="more-slides" class="titleslide slide level1"><h1>More slides</h1></section></section>
<section><section id="single-molecule-sequencing" class="titleslide slide level1"><h1>Single molecule sequencing</h1></section><section id="pacbio-rs-ii" class="slide level2">
<h1>PacBio RS II</h1>
<p><a href="http://www.pacificbiosciences.com/"><img src="images/PacBio-RS-II.png" alt="PacBio RS II" /></a></p>
</section><section id="oxford-nanopore-minion" class="slide level2">
<h1>Oxford Nanopore MinION</h1>
<p><a href="https://www.nanoporetech.com/"><img src="images/MinION.jpg" alt="Oxford Nanopore MinION" /></a></p>
</section><section id="sequencing-technologies" class="slide level2">
<h1>Sequencing technologies</h1>
<center>
<table>
<thead>
<tr class="header">
<th style="text-align: left;">Technology</th>
<th style="text-align: left;">Read length</th>
<th style="text-align: left;">Error rate</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">Sanger</td>
<td style="text-align: left;">800 bp</td>
<td style="text-align: left;">0.1–1%</td>
</tr>
<tr class="even">
<td style="text-align: left;">454</td>
<td style="text-align: left;">700 bp</td>
<td style="text-align: left;">~1%</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Illumina</td>
<td style="text-align: left;">2 x 300 bp</td>
<td style="text-align: left;">~0.1%</td>
</tr>
<tr class="even">
<td style="text-align: left;">PacBio</td>
<td style="text-align: left;">8–40 kbp</td>
<td style="text-align: left;">~13%</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Oxford Nanopore</td>
<td style="text-align: left;">8–200 kbp</td>
<td style="text-align: left;">~15%</td>
</tr>
</tbody>
</table>
</center>
</section><section id="a-visual-comparison" class="slide level2">
<h1>A visual comparison</h1>
<figure>
<img src="images/sms-igv.png" alt="8 kbp MinION read and 2 x 300 bp Illumina read" /><figcaption>8 kbp MinION read and 2 x 300 bp Illumina read</figcaption>
</figure>
</section><section id="pacbio-circular-consensus" class="slide level2">
<h1>PacBio circular consensus</h1>
<p><a href="http://files.pacb.com/software/smrtanalysis/2.2.0/doc/smrtportal/help/!SSL!/Webhelp/Portal_PacBio_Glossary.htm"><img src="images/PacBio-CCS.png" alt="Circular consensus" /></a></p>
</section><section id="nanopore-2d" class="slide level2">
<h1>Nanopore 2D</h1>
<p><a href="http://www.nature.com/nmeth/journal/v12/n4/fig_tab/nmeth.3290_SF13.html"><img src="images/Nanopore-2D.png" alt="Nanopore 2D reads" /></a></p>
</section></section>
<section><section id="assembly-of-single-molecule-sequencing" class="titleslide slide level1"><h1>Assembly of Single-molecule Sequencing</h1></section><section id="assembly-stages" class="slide level2">
<h1>Assembly Stages</h1>
<h3 id="overlap-1">Overlap</h3>
<p>Find all significantly overlapping reads</p>
<h3 id="correct">Correct</h3>
<p>Recall the consensus base of each read</p>
<h3 id="layout-1">Layout</h3>
<p>Determine the order and orientation of the reads</p>
<h3 id="consensus-1">Consensus</h3>
<p>Call the consensus base of each contig</p>
</section><section id="tools" class="slide level2">
<h1>Tools</h1>
<h3 id="overlap-2">Overlap</h3>
<p><a href="https://github.com/PacificBiosciences/blasr">BLASR</a> · <a href="https://github.com/thegenemyers/DALIGNER">DALIGNER</a> · <a href="https://github.com/marbl/MHAP">MHAP</a></p>
<h3 id="correct-1">Correct</h3>
<p><a href="https://github.com/PacificBiosciences/pbdagcon">PBDAGCon</a> · <a href="https://github.com/PacificBiosciences/falcon">Falcon</a> · <a href="https://dazzlerblog.wordpress.com/">Dazzler</a> · <a href="https://github.com/jts/nanocorrect">Nanocorrect</a></p>
<h3 id="layout-2">Layout</h3>
<p><a href="http://wgs-assembler.sourceforge.net/">Celera Assembler</a> · <a href="https://github.com/PacificBiosciences/falcon">Falcon</a> · <a href="https://dazzlerblog.wordpress.com/">Dazzler</a></p>
<h3 id="consensus-2">Consensus</h3>
<p><a href="https://github.com/PacificBiosciences/GenomicConsensus">Quiver</a> · <a href="https://github.com/jts/nanopolish">Nanopolish</a></p>
</section><section id="pipelines" class="slide level2">
<h1>Pipelines</h1>
<table>
<thead>
<tr class="header">
<th style="text-align: left;">Assembler</th>
<th style="text-align: left;">Overlap</th>
<th style="text-align: left;">Correct</th>
<th style="text-align: left;">Layout</th>
<th style="text-align: left;">Consensus</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;"><a href="https://github.com/PacificBiosciences/Bioinformatics-Training/wiki/HGAP-in-SMRT-Analysis">HGAP</a></td>
<td style="text-align: left;">BLASR</td>
<td style="text-align: left;">PBDAGCon</td>
<td style="text-align: left;">Celera</td>
<td style="text-align: left;">Quiver</td>
</tr>
<tr class="even">
<td style="text-align: left;"><a href="https://github.com/PacificBiosciences/falcon">Falcon</a></td>
<td style="text-align: left;">DALIGNER</td>
<td style="text-align: left;">Falcon</td>
<td style="text-align: left;">Falcon</td>
<td style="text-align: left;">Quiver</td>
</tr>
<tr class="odd">
<td style="text-align: left;"><a href="http://www.cbcb.umd.edu/software/PBcR/">PBcR</a></td>
<td style="text-align: left;">MHAP</td>
<td style="text-align: left;">Falcon</td>
<td style="text-align: left;">Celera</td>
<td style="text-align: left;">Quiver</td>
</tr>
<tr class="even">
<td style="text-align: left;"><a href="https://dazzlerblog.wordpress.com/">Dazzler</a></td>
<td style="text-align: left;">DALIGNER</td>
<td style="text-align: left;">Dazzler</td>
<td style="text-align: left;">Dazzler</td>
<td style="text-align: left;">Quiver</td>
</tr>
<tr class="odd">
<td style="text-align: left;"><a href="http://schatzlab.cshl.edu/data/nanocorr/">Nanocorr</a></td>
<td style="text-align: left;">BLAST</td>
<td style="text-align: left;">PBDAGCon</td>
<td style="text-align: left;">Celera</td>
<td style="text-align: left;">Celera</td>
</tr>
<tr class="even">
<td style="text-align: left;"><a href="https://github.com/jts/nanopolish">Nanopolish</a></td>
<td style="text-align: left;">DALIGNER</td>
<td style="text-align: left;">Nanocorrect</td>
<td style="text-align: left;">Celera</td>
<td style="text-align: left;">Nanopolish</td>
</tr>
</tbody>
</table>
<p><a href="https://github.com/sjackman/thesis-committee/blob/gh-pages/images/sms-olc.gv"><img src="images/sms-olc.png" alt="SMS assembly tools" /></a></p>
</section><section id="assembling-large-genomes" class="slide level2">
<h1>Assembling large genomes</h1>
<p><a href="https://github.com/PacificBiosciences/Bioinformatics-Training/wiki/Large-Genome-Assembly-with-PacBio-Long-Reads"><img src="images/PacBio-LargeGenomes.png" alt="PacBio-LargeGenomes" /></a></p>
</section><section id="assembling-small-genomes-in-2015" class="slide level2">
<h1>Assembling small genomes in 2015</h1>
<h3 id="nanopore-illumina">Nanopore + Illumina</h3>
<p><a href="http://schatzlab.cshl.edu/data/nanocorr/">Nanocorr</a> <em>Saccharomyces cerevisiae</em> (12 Mbp)</p>
<h3 id="nanopore-only">Nanopore only</h3>
<p><a href="https://github.com/jts/nanopolish">Nanopolish</a> <em>Escherichia coli</em> (5 Mbp)</p>
</section></section>
<section><section id="retool-for-sms" class="titleslide slide level1"><h1>Retool for SMS</h1></section><section id="retool-for-sms-1" class="slide level2">
<h1>Retool for SMS</h1>
<h3 id="alignment">Alignment</h3>
<p>Seed and extend is difficult when<br />
one in seven bases is incorrect</p>
<h3 id="assembly">Assembly</h3>
<p>de Bruijn graphs also require accurate seeds<br />
Return to overlap, layout, consensus (OLC)</p>
</section><section id="align" class="slide level2">
<h1>Align</h1>
<ul>
<li><a href="https://github.com/PacificBiosciences/blasr">BLASR</a> PacBio</li>
<li><a href="https://github.com/lh3/bwa">BWA-MEM</a> (Burrows-Wheeler Aligner) Heng Li</li>
<li><a href="http://last.cbrc.jp/">LAST</a> Computational Biology Research Consortium, Tokyo, Japan</li>
</ul>
</section><section id="overlap-3" class="slide level2">
<h1>Overlap</h1>
<ul>
<li><a href="https://github.com/PacificBiosciences/blasr">BLASR</a> PacBio</li>
<li><a href="https://github.com/thegenemyers/DALIGNER">DALIGNER</a> Gene Myers</li>
<li><a href="https://github.com/marbl/MHAP">MHAP</a> (MinHash Alignment Process) Adam Phillippy</li>
</ul>
</section><section id="read-correction" class="slide level2">
<h1>Read correction</h1>
<ul>
<li><a href="https://github.com/PacificBiosciences/pbdagcon">PBDAGCon</a> (PacBio DAG Consensus) PacBio</li>
<li><a href="https://github.com/PacificBiosciences/falcon">Falcon</a> PacBio</li>
<li><a href="https://dazzlerblog.wordpress.com/">Dazzler</a> (Dresden Azzembler) Gene Myers</li>
<li><a href="https://github.com/jts/nanocorrect">Nanocorrect</a> Jared Simpson (using <a href="https://sourceforge.net/projects/poamsa/">POA</a>)</li>
</ul>
</section><section id="layout-3" class="slide level2">
<h1>Layout</h1>
<ul>
<li><a href="http://wgs-assembler.sourceforge.net/">Celera Assembler</a> J. Craig Venter Institute (JCVI)</li>
<li><a href="https://github.com/PacificBiosciences/falcon">Falcon</a> PacBio</li>
<li><a href="https://dazzlerblog.wordpress.com/">Dazzler</a> (Dresden Azzembler) Gene Myers</li>
</ul>
</section><section id="consensus-3" class="slide level2">
<h1>Consensus</h1>
<ul>
<li><a href="https://github.com/PacificBiosciences/GenomicConsensus">Quiver</a> PacBio</li>
<li><a href="https://github.com/jts/nanopolish">Nanopolish</a> Jared Simpson</li>
</ul>
</section><section id="assembly-pipeline" class="slide level2">
<h1>Assembly Pipeline</h1>
<ul>
<li><a href="https://github.com/PacificBiosciences/Bioinformatics-Training/wiki/HGAP-in-SMRT-Analysis">HGAP</a> (Hierarchical Genome Assembly Process) PacBio</li>
<li><a href="https://github.com/PacificBiosciences/falcon">Falcon</a> PacBio</li>
<li><a href="http://www.cbcb.umd.edu/software/PBcR/">PBcR</a> (PacBio Corrected Reads) Celera Assembler, JCVI</li>
<li><a href="https://dazzlerblog.wordpress.com/">Dazzler</a> (Dresden Azzembler) Gene Myers</li>
<li><a href="http://schatzlab.cshl.edu/data/nanocorr/">Nanocorr</a> Mike Schatz</li>
<li><a href="https://github.com/jts/nanopolish">Nanopolish</a> Jared Simpson</li>
</ul>
</section></section>
    </div>
  </div>


  <script src="reveal.js/lib/js/head.min.js"></script>
  <script src="reveal.js/js/reveal.js"></script>

  <script>

      // Full list of configuration options available here:
      // https://github.com/hakimel/reveal.js#configuration
      Reveal.initialize({
        controls: true,         // Display controls in the bottom right corner
        progress: true,         // Display a presentation progress bar
        history: true,          // Push each slide change to the browser history
        center: true,                       // Vertical centering of slides
        maxScale: 1.5,                  // Bounds for smallest/largest possible content scale
        slideNumber: true,                // Display the page number of the current slide
        theme: 'sky', // available themes are in /css/theme
        transition: Reveal.getQueryHash().transition || 'default', // default/cube/page/concave/zoom/linear/fade/none

        // Optional libraries used to extend on reveal.js
        dependencies: [
          { src: 'reveal.js/lib/js/classList.js', condition: function() { return !document.body.classList; } },
          { src: 'reveal.js/plugin/zoom-js/zoom.js', async: true, condition: function() { return !!document.body.classList; } },
          { src: 'reveal.js/plugin/notes/notes.js', async: true, condition: function() { return !!document.body.classList; } },
//          { src: 'reveal.js/plugin/search/search.js', async: true, condition: function() { return !!document.body.classList; }, }
//          { src: 'reveal.js/plugin/remotes/remotes.js', async: true, condition: function() { return !!document.body.classList; } }
]});
    </script>
    </body>
</html>
