<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8">
  <meta name="generator" content="pandoc">
  <meta name="author" content="Shaun Jackman">
  <meta name="dcterms.date" content="2018-01-30">
  <title>PhD thesis committee meeting</title>
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black-translucent">
  <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no, minimal-ui">
  <link rel="stylesheet" href="reveal.js/css/reveal.css">
  <style type="text/css">code{white-space: pre;}</style>
  <link rel="stylesheet" href="reveal.js/css/theme/sky.css" id="theme">
  <!-- Printing and PDF exports -->
  <script>
    var link = document.createElement( 'link' );
    link.rel = 'stylesheet';
    link.type = 'text/css';
    link.href = window.location.search.match( /print-pdf/gi ) ? 'reveal.js/css/print/pdf.css' : 'reveal.js/css/print/paper.css';
    document.getElementsByTagName( 'head' )[0].appendChild( link );
  </script>
  <!--[if lt IE 9]>
  <script src="reveal.js/lib/js/html5shiv.js"></script>
  <![endif]-->
</head>
<body>
  <div class="reveal">
    <div class="slides">

<section>
  <h1 class="title">PhD thesis committee meeting</h1>
  <p class="author">Shaun Jackman</p>
  <p class="date">2018-01-30</p>
</section>

<section id="shaun-jackman" class="slide level2">
<h2>Shaun Jackman</h2>
<div style="white-space: pre-line;"><a href="http://bcgsc.ca">Genome Sciences Centre</a>, BC Cancer, Vancouver, Canada
<a href="https://twitter.com/sjackman">@sjackman</a> · <a href="https://github.com/sjackman">github.com/sjackman</a> · <a href="http://sjackman.ca">sjackman.ca</a></div>
<p><a href="http://sjackman.ca"><img src="images/sjackman.jpg" alt="Photo" /></a></p>
</section>
<section id="thesis-committee" class="slide level2">
<h2>Thesis Committee</h2>
<div style="white-space: pre-line;"><a href="http://www.bcgsc.ca/faculty/inanc-birol">Inanc Birol</a>
<a href="http://bohlmannlab.msl.ubc.ca/">Joerg Bohlmann</a>
<a href="http://www.stat.ubc.ca/~jenny/">Jenny Bryan</a>
<a href="http://hallam.microbiology.ubc.ca/">Steven Hallam</a></div>
</section>
<section id="timeline" class="slide level2">
<h2>Timeline</h2>
<ul>
<li>Previous meeting 2015-07-27</li>
<li>Fourth committee meeting 2018-01-30</li>
<li>Start writing early 2018</li>
<li>Submit thesis and defend late 2018</li>
<li>Graduate spring 2019</li>
</ul>
</section>
<section id="conferences" class="slide level2">
<h2>Conferences</h2>
<ul>
<li>Three talks and four posters</li>
<li><a href="http://sjackman.ca/linuxbrew-slides">ISMB 2016 BOSC talk</a> and <a href="https://f1000research.com/posters/5-1795">poster</a><br />
<small>Linuxbrew and Homebrew for Cross-platform Package Management</small></li>
<li><a href="https://f1000research.com/posters/5-2072">ISMB 2016 HiTSeq poster</a><br />
<small>Organellar Genomes of White spruce (<em>Picea glauca</em>): Assembly and annotation</small></li>
<li><a href="http://sjackman.ca/picea-sitchensis-organelles-slides/">PAG 2017 talk</a> and <a href="https://f1000research.com/posters/6-132">AGBT 2017 poster</a><br />
<small>Organellar Genomes of Sitka Spruce (<em>Picea sitchensis</em>): Assembly and Annotation</small></li>
<li><a href="http://sjackman.ca/tigmint-slides/">ISMB 2017 talk</a> and <a href="https://f1000research.com/posters/6-1406">poster</a><br />
<small>Tigmint: Correct Misassemblies Using Linked Reads From Large Molecules</small></li>
</ul>
</section>
<section id="teaching-assistant" class="slide level2">
<h2>Teaching Assistant</h2>
<p>UBC Master of Data Science</p>
<ul>
<li>2016 <a href="https://ubc-mds.github.io/course-descriptions/DSCI_522_dsci-workflows/">DSCI 522</a> Data Science Workflows</li>
<li>2017 <a href="https://ubc-mds.github.io/course-descriptions/DSCI_531_viz-1/">DSCI 531</a> Data Visualization I</li>
</ul>
</section>
<section class="slide level2">

<div style="white-space: pre-line;">2017 Genome Research
<a href="https://doi.org/10.1101/gr.214346.116"><img src="images/ABySS-2.0-2017.png" alt="ABySS 2.0" height="150" /></a>
2016 Genome Biology and Evolution
<a href="https://doi.org/10.1093/gbe/evv244"><img src="images/white-spruce-organelles.png" alt="Organellar Genomes of White Spruce" height="150" /></a>
2015 PLOS ONE
<a href="https://doi.org/10.1371/journal.pone.0128026"><img src="images/UniqTag-PLOSONE-small.png" alt="UniqTag" height="150" /></a></div>
<aside class="notes">
<ul>
<li>ABySS 2.0: resource-efficient assembly of large genomes using a Bloom filter<br />
<small> <strong>SD Jackman</strong><sup>*</sup>, BP Vandervalk<sup>*</sup>, H Mohamadi, J Chu, S Yeo, SA Hammond, …<br />
Genome Research 2017 </small></li>
<li>Organellar genomes of white spruce (<em>Picea glauca</em>): assembly and annotation<br />
<small> <strong>SD Jackman</strong>, RL Warren, EA Gibb, BP Vandervalk, H Mohamadi, J Chu, …<br />
Genome Biology and Evolution 2015 </small></li>
<li>UniqTag: Content-derived unique and stable identifiers for gene annotation<br />
<small> <strong>SD Jackman</strong>, J Bohlmann, I Birol<br />
PLOS ONE 2015 </small></li>
</ul>
</aside>
</section>
<section id="papers" class="slide level2">
<h2>Papers</h2>
<ul>
<li>Assembly of the complete Sitka spruce chloroplast genome using 10X Genomics’ GemCode sequencing data <small> L Coombe, RL Warren, <strong>SD Jackman</strong>, C Yang, BP Vandervalk, RA Moore, …<br />
PloS one 2016 </small></li>
<li>Spaced seed data structures for de novo assembly<br />
<small> I Birol, J Chu, H Mohamadi, <strong>SD Jackman</strong>, K Raghavan, BP Vandervalk, …<br />
International journal of genomics 2015 </small></li>
<li>Konnector v2.0: pseudo-long reads from paired-end sequencing data<br />
<small> BP Vandervalk, C Yang, Z Xue, K Raghavan, J Chu, H Mohamadi, <strong>SD Jackman</strong>,…<br />
BMC medical genomics 2015 </small></li>
<li>Sealer: a scalable gap-closing application…<br />
<small> D Paulino, RL Warren, BP Vandervalk, A Raymond, <strong>SD Jackman</strong>, I Birol<br />
BMC Bioinformatics 2015 </small></li>
<li>Improved white spruce (<em>Picea glauca</em>) genome…<br />
<small> RL Warren, CI Keeling, MMS Yuen, A Raymond, GA Taylor, …<br />
The Plant Journal 2015 </small></li>
</ul>
</section>
<section id="papers-1" class="slide level2">
<h2>Papers</h2>
<ul>
<li>Three first-author (or joint) papers since 2015</li>
<li>Collaborated on 29 papers since 2009</li>
<li>25 papers with at least 10 citations</li>
<li>Two first-author manuscripts in preparation</li>
</ul>
</section>
<section id="manuscripts" class="slide level2">
<h2>Manuscripts</h2>
<ul>
<li>ORCA: A Comprehensive Bioinformatics Container Environment for Education and Research <small> <strong>SD Jackman</strong><sup>*</sup>, T Mozgacheva<sup>*</sup>, B O’Huiginn, L Bailey, I Birol, SJM Jones </small></li>
<li>Tigmint: Correct Assembly Errors Using Linked Reads From Large Molecules <small> <strong>SD Jackman</strong>, J Chu, RL Warren, BP Vandervalk, L Coombe, S Yeo, … </small></li>
</ul>
<p><a href="http://github.com/bcgsc/tigmint"><img src="images/tigmint.png" alt="Tigmint" height="250" /></a></p>
</section>
<section id="thesis-outline" class="slide level2">
<h2>Thesis Outline</h2>
<h3 id="methods">Methods</h3>
<ol type="1">
<li>ABySS 2.0: resource-efficient assembly of large genomes</li>
<li>Tigmint: Correcting misassemblies using linked reads</li>
<li>UniqTag: Content-derived unique and stable identifiers</li>
</ol>
<h3 id="genomes">Genomes</h3>
<ol type="1">
<li>Organellar genomes of white spruce (<em>Picea glauca</em>)<br />
using paired-end and mate-pair reads</li>
<li>Organellar genomes of Sitka spruce (<em>Picea sitchensis</em>)<br />
using linked reads and Nanopore reads</li>
<li>Genome assembly of western redcedar (<em>Thuja plicata</em>)<br />
using paired-end, mate-pair, and linked reads</li>
</ol>
</section>
<section id="white-spruce-organelles" class="slide level2">
<h2>White Spruce Organelles</h2>
<ul>
<li>Assembled cpDNA and mtDNA genomes</li>
<li>Annotated genes (mRNA, rRNA, tRNA) and repeats</li>
<li>Analysed RNA-seq data to quantify
<ul>
<li>transcript abundance in eight tissues</li>
<li>expressed ORFs</li>
<li>C-to-U RNA editing</li>
<li>cryptic ACG start codons due to C-to-U RNA editing</li>
</ul></li>
<li>Submitted annotated genomes to GenBank</li>
<li>Published paper in Genome Biology and Evolution (2015)</li>
</ul>
</section>
<section class="slide level2">

<figure>
<img src="images/pgmt-genes.png" alt="White Spruce Mitochondrion" height="600" /><figcaption>White Spruce Mitochondrion</figcaption>
</figure>
</section>
<section id="sitka-spruce-plastid" class="slide level2">
<h2>Sitka Spruce Plastid</h2>
<ul>
<li>Assembled by Rene Warren and Lauren Coombe<br />
using linked reads</li>
<li>Annotated genes (mRNA, rRNA, tRNA)</li>
<li>Submitted annotated genomes to GenBank</li>
<li>Published paper in PLOS ONE (2016)</li>
</ul>
</section>
<section class="slide level2">

<div style="white-space: pre-line;"><img src="images/picea-sitchensis-plastid.png" alt="Sitka spruce plastid" height="600" /></div>
</section>
<section id="sitka-spruce-mitochondrion" class="slide level2">
<h2>Sitka Spruce Mitochondrion</h2>
<ul>
<li>10x Genomics Chromium sequencing</li>
<li>&gt; 50x mitochondrial coverage in one lane</li>
<li>7 lanes of Oxford Nanopore Sequencing</li>
<li>~ 1x nuclear coverage</li>
<li>&gt; 10x mitochondrial coverage</li>
<li>Assemble Nanopore reads</li>
<li>Scaffold and polish with linked reads</li>
</ul>
</section>
<section class="slide level2">

<figure>
<img src="images/sitka-spruce-mt-miniasm.png" alt="Nanopore reads assembled with Miniasm" /><figcaption>Nanopore reads assembled with Miniasm</figcaption>
</figure>
</section>
<section id="sitka-spruce-mitochondrion-1" class="slide level2">
<h2>Sitka Spruce Mitochondrion</h2>
<ul>
<li>Complete the genome assembly</li>
<li>Determine chromosomal structure</li>
<li>Annotate the genome</li>
<li>Submit the genome to GenBank</li>
<li>Write the manuscript</li>
</ul>
</section>
<section id="abyss-2.0" class="slide level2">
<h2>ABySS 2.0</h2>
<ul>
<li>Implemented Bloom filter de Bruijn Graph
<ul>
<li><a href="http://doi.org/10.1109/BIBM.2014.6999126">Konnector</a> and <a href="https://doi.org/10.1186/1755-8794-8-S3-S1">Konnector 2.0</a> with Ben Vandervalk</li>
<li><a href="http://www.biomedcentral.com/1471-2105/16/230">Sealer</a> with Daniel Paulino</li>
<li><a href="#/abyss-2.0-1">ABySS 2.0</a> with Ben Vandervalk</li>
</ul></li>
<li>Reduce memory usage by twelve fold over ABySS 1.0</li>
<li>Assemble a conifer genome with a single machine</li>
<li>Memory usage is independent of parameter <em>k</em></li>
<li>Assembled a human genome with ABySS 2.0 (35 GB RAM)</li>
</ul>
</section>
<section id="abyss-2.0-1" class="slide level2">
<h2>ABySS 2.0</h2>
<ul>
<li>Compared to ABySS 1.5 and six other assemblers</li>
<li>Submitted genome assemblies to NCBI</li>
<li>Published paper in Genome Research (2017)</li>
<li>Submitted paper to RECOMB 2018 Highlights</li>
</ul>
<div style="white-space: pre-line;"><img src="http://sjackman.ca/img/abyss-2.0.png" alt="ABySS 2.0 chromosomes" /></div>
</section>
<section id="tigmint" class="slide level2">
<h2>Tigmint</h2>
<p>Correcting misassemblies using linked reads</p>
<ul>
<li>Incorrectly assembled sequence complicates all downstream analyses</li>
<li>Misassemblies also limit contiguity</li>
<li>Cut contigs where linked reads and assembly disagree</li>
<li>Tigmint + ARCS improved contiguity by half over ARCS alone in human</li>
<li>Further developing the tool with Lauren Coombe</li>
<li>Submit manuscript to RECOMB-Seq 2018</li>
</ul>
<p><a href="http://github.com/bcgsc/tigmint"><img src="http://sjackman.ca/img/tigmint.png" alt="Tigmint" height="100" /></a></p>
</section>
<section class="slide level2">

<div style="white-space: pre-line;"><img src="images/tigmint-jupiter.png" alt="Tigmint Jupiter plot" /></div>
</section>
<section class="slide level2">

<div style="white-space: pre-line;"><img src="images/tigmint-metrics.png" alt="Tigmint Assembly Metrics" height="600" /></div>
</section>
<section id="western-redcedar-thuja-plicata" class="slide level2">
<h2>Western Redcedar (<em>Thuja plicata</em>)</h2>
<h3 id="method">Method</h3>
<ul>
<li>Trim adapters with Trimadap and NxTrim</li>
<li>Count k-mers with ntCard</li>
<li>Estimate genome size GenomeScope</li>
<li>Assemble PE and MP reads with ABySS 2.0</li>
<li>Scaffold with Chromium reads using ARCS</li>
<li>Assess genome completeness using BUSCO</li>
</ul>
<h3 id="results">Results</h3>
<ul>
<li>Scaffold NG50 of 1.2 Mbp</li>
<li>BUSCO 54% complete, 7% fragmented</li>
</ul>
</section>
<section id="thesis-outline-1" class="slide level2">
<h2>Thesis Outline</h2>
<h3 id="methods-1">Methods</h3>
<ol type="1">
<li>ABySS 2.0: resource-efficient assembly of large genomes</li>
<li>Tigmint: Correcting misassemblies using linked reads</li>
<li>UniqTag: Content-derived unique and stable identifiers</li>
</ol>
<h3 id="genomes-1">Genomes</h3>
<ol type="1">
<li>Organellar genomes of white spruce (<em>Picea glauca</em>)<br />
using paired-end and mate-pair reads</li>
<li>Organellar genomes of Sitka spruce (<em>Picea sitchensis</em>)<br />
using linked reads and Nanopore reads</li>
<li>Genome assembly of western redcedar (<em>Thuja plicata</em>)<br />
using paired-end, mate-pair, and linked reads</li>
</ol>
</section>
<section><section id="fin" class="titleslide slide level1"><h1>fin</h1></section><section id="shaun-jackman-1" class="slide level2">
<h2>Shaun Jackman</h2>
<div style="white-space: pre-line;"><a href="http://bcgsc.ca">Genome Sciences Centre</a>, BC Cancer, Vancouver, Canada
<a href="https://twitter.com/sjackman">@sjackman</a> · <a href="https://github.com/sjackman">github.com/sjackman</a> · <a href="http://sjackman.ca">sjackman.ca</a></div>
<p><a href="http://sjackman.ca"><img src="images/sjackman.jpg" alt="Photo" /></a></p>
</section></section>
    </div>
  </div>

  <script src="reveal.js/lib/js/head.min.js"></script>
  <script src="reveal.js/js/reveal.js"></script>

  <script>

      // Full list of configuration options available at:
      // https://github.com/hakimel/reveal.js#configuration
      Reveal.initialize({
        // Display the page number of the current slide
        slideNumber: true,
        // Push each slide change to the browser history
        history: true,

        // Optional reveal.js plugins
        dependencies: [
          { src: 'reveal.js/lib/js/classList.js', condition: function() { return !document.body.classList; } },
          { src: 'reveal.js/plugin/zoom-js/zoom.js', async: true },
              { src: 'reveal.js/plugin/notes/notes.js', async: true }
        ]
      });
    </script>
    </body>
</html>
